modules/sso/sso-pre-launch-checks/added/sso_image_pre_launch_checks.sh

```diff
@@ -75,3 +76,17 @@ function verify_CIAM_1757_fix_present() {
     exit "${errorExitCode}"
   fi
 }
+
+# CIAM-1975
+#
+# Verify one-off patch for CIAM-1975 got properly installed to the expected location
+#
+function verify_CIAM_1975_fix_present() {
+  local -r errorExitCode="1"
+  if ! find "${JBOSS_HOME}"/modules/system/layers -name '*-rhsso-1974.jar' 2> /dev/null | grep -q .
```

Comment:
I'm not totally sure. I would suspect the lack of quotes around the subshell as the problem, but the command behaves as expected when I tested it (file `labels` exists, `zabels` does not):

```
[aszczucz@tyrfing tmp]$ [ -z "$(find -name 'labels' 2>/dev/null)" ]; echo $?
1
[aszczucz@tyrfing tmp]$ [ -z $(find -name 'labels' 2>/dev/null) ]; echo $?
1
[aszczucz@tyrfing tmp]$ [ -z $(find -name 'zabels' 2>/dev/null) ]; echo $?
0
[aszczucz@tyrfing tmp]$ [ -z "$(find -name 'zabels' 2>/dev/null)" ]; echo $?
0
```

I was trying to think of an alternative, but find itself doesn't seem capable of returning an exit code based on finding the files or not. One I came up with is `xargs -r false`, but that's similar in complexity to the `grep -q .`

```
[aszczucz@tyrfing tmp]$ find -name 'zabels' 2>/dev/null | xargs -r false; echo $?
0
[aszczucz@tyrfing tmp]$ find -name 'labels' 2>/dev/null | xargs -r false; echo $?
123
```

Reply:
@ASzc 

Yes, it seems to have been caused by the lack of quotes around the whole subshell. For `-z` it doesn't seem to matter (both alternatives return same results for a non-existing file), but for `-n` it seems to matter:
```
[iankko@localhost tmp]$ ls -l /tmp/non_existing_file
ls: cannot access '/tmp/non_existing_file': No such file or directory
[iankko@localhost tmp]$ [ -z $(find "/tmp" -name non_existing_file 2> /dev/null) ]; echo $?
0
[iankko@localhost tmp]$ [ -z "$(find /tmp -name non_existing_file 2> /dev/null)" ]; echo $?
0
[iankko@localhost tmp]$ [ -n $(find "/tmp" -name non_existing_file 2> /dev/null) ]; echo $?
0
[iankko@localhost tmp]$ [ -n "$(find /tmp -name non_existing_file 2> /dev/null)" ]; echo $?
1
```

Don't remember anymore, when preparing the change, which of the variants I tried / intended to use, but given the unclear output above, the  `grep .` pipe after find seemed to be the most reasonable / reliable one to me.

As for the alternatives, yes AFAICT some stackoverflow posts seemed to suggest to add `-exec` option to determine if the file existed or not (since it's not possible to determine using just exit code of `find`), but that variant / version seemed even more unreadable to me, than the `| grep .` one.